src/coreComponents/physicsSolvers/CMakeLists.txt

```diff
@@ -115,7 +146,35 @@ endif()
 if( GEOS_ENABLE_MULTIPHYSICS )
   add_subdirectory( multiphysics )
   list( APPEND dependencyList multiPhysicsSolvers )
+  list( APPEND kernelTemplateFileList 
+        multiphysics/poromechanicsKernels/PoromechanicsKernels.cpp.template
+        multiphysics/poromechanicsKernels/PoromechanicsEFEMKernels.cpp.template
+        multiphysics/poromechanicsKernels/ThermoPoromechanicsKernels.cpp.template
+        multiphysics/poromechanicsKernels/PoromechanicsDamageKernels.cpp.template )
+
+endif()
+
+if(NOT DEFINED GEOS_KERNEL_SPEC)
+  set(GEOS_KERNEL_SPEC "${CMAKE_CURRENT_LIST_DIR}/kernels-all.json" )
 endif()
+  
+file( READ ${GEOS_KERNEL_SPEC} kernelSpecs )
+
+ 
+foreach( kernelTemplateFile ${kernelTemplateFileList} )
+  get_filename_component( jsonKey ${kernelTemplateFile} NAME_WE )
+  generateKernels( TEMPLATE ${kernelTemplateFile}
+                   JSON kernelSpecs
+                   KEY ${jsonKey}
+                   HEADERS headerFiles
+                   SOURCES sourceFiles )
+
+  list(APPEND physicsSolvers_headers ${headerFiles})
+  list(APPEND physicsSolvers_sources ${sourceFiles})
```

Comment:
I had not noticed but in reality the kernels should be added to the sources of each individual package not to the physicsSolvers. We may want to fix this.

Reply:
@CusiniM I pushed everything down into the components